pkgs/build-support/setup-hooks/patch-shebangs.sh

```diff
@@ -54,7 +54,11 @@ patchShebangs() {
                 echo "$f: interpreter directive changed from \"$oldInterpreterLine\" to \"$newInterpreterLine\""
                 # escape the escape chars so that sed doesn't interpret them
                 escapedInterpreterLine=$(echo "$newInterpreterLine" | sed 's|\\|\\\\|g')
+                # Preserve times, see: https://github.com/NixOS/nixpkgs/pull/33084
+                cp -p "$f" "$f.tmp"
```

Comment:
Making a full copy of the file, just to store its mtime, seems very inefficient to me.

Can't you do `touch -r "$f" "$f.tmp"`?

Reply:
That file does not exists before `cp` -> sed edit $f inplace.

Reply:
`touch` will create it.

Reply:
right.